### PR TITLE
chore: upgrade to biome@1.7.2

### PR DESCRIPTION
**Pull request by bobheadxi**
Upgrades Biome, as Data Team wants to use https://biomejs.dev/linter/rules/no-restricted-imports/, which is not available in the version of Biome we are currently using.

It seems like quite a lot has changed within Biome - I've migrated the rules that have been promoted out of `nursery` into their modern equivalents.

## Test plan

`pnpm run biome` exits with status 0 locally, required CI passes

### CODE REVIEW

File: vscode/src/completions/text-processing/parse-completion.ts [hunk 1]
```diff
@@ -111,7 +111,6 @@ function pasteCompletion(params: PasteCompletionParams): PasteCompletionResult {
         docContext: {
             position,
             currentLineSuffix,
-            // biome-ignore lint/nursery/noInvalidUseBeforeDeclaration: it's actually correct
             positionWithoutInjectedCompletionText = position,
```
Comment (bobheadxi) [inline vscode/src/completions/text-processing/parse-completion.ts:114]: No longer an error

File: vscode/webviews/Components/FileLink.tsx [hunk 2]
```diff
@@ -71,7 +71,6 @@ export const FileLink: React.FunctionComponent<FileLinkProps & { className?: str
     return (
         <div className={classNames(styles.linkContainer, className)}>
             {isTooLarge && <i className="codicon codicon-warning" title={WARNING} />}
-            {/* biome-ignore lint/a11y/useValidAnchor: The onClick handler is only used for logging */}
```
Comment (bobheadxi) [inline vscode/webviews/Components/FileLink.tsx:74]: No longer an error

File: vscode/webviews/OnboardingExperiment.tsx [hunk 3]
```diff
@@ -40,7 +40,6 @@ const WebLogin: React.FunctionComponent<
                 </a>
             </li>
             <li>
-                {/* biome-ignore lint/a11y/useValidAnchor: can fix with a lot of CSS but not a priority */}
```
Comment (bobheadxi) [inline vscode/webviews/OnboardingExperiment.tsx:43]: No longer an error

File: vscode/webviews/SearchPanel.tsx [hunk 4]
```diff
@@ -269,6 +269,7 @@ export const SearchPanel: React.FunctionComponent<{ vscodeAPI: VSCodeWrapper }>
             <div className={styles.searchResultsContainer}>
                 {results.map((result, fileIndex) => (
                     <>
+                        {/* biome-ignore lint/correctness/useJsxKeyInIterable: Each of the possible divs below have an appropriate key. */}
                         {/* File result */}
                         <div
```
Comment (bobheadxi) [inline vscode/webviews/SearchPanel.tsx:274]: I think Biome expects a `<div>` with a unique key wrapping the options below - I'm worried about messing up any styling here, and each of the possible divs below have keys

**Review by sqs — APPROVED** [review]

**Comment by philipp-spiess** [comment]
FYI this commit on `main` started to break `biome` for us :( see https://sourcegraph.slack.com/archives/C05AGQYD528/p1714639734553859

**Pull request merged**